docs/rules/newline-before-return.md

```diff
@@ -0,0 +1,122 @@
+# Require newline before `return` statement (newline-before-return)
+
+There is no hardfast rule about whether newlines should precede `return` statements in JavaScript. However, clearly delineating where a function is returning can greatly increase the readability and clarity of the code. For example:
+
+```js
+function foo(bar) {
+  var baz = 'baz';
+  if (!bar) {
+    bar = baz;
+    return bar;
+  }
+  return bar;
+}
+```
+
+
+Adding newlines visibly separates the return statements from the previous lines, making it clear where the function exits and what value it returns:
+
+```js
+function foo(bar) {
+  var baz = 'baz';
+
+  if (!bar) {
+    bar = baz;
+
+    return bar;
+  }
+
+  return bar;
+}
+```
+
+## Rule Details
+
+This rule aims to increase code clarity by requiring a blank line before `return` statements, except when the `return` is alone inside a statement group (such as an if statement). In the latter case, the `return` statement does not need to be delineated by virtue of it being alone.
+
+When this rule is on, the follow patterns are considered problems:
+
+```js
+/*eslint newline-before-return: 2*/
+
+function foo() {
+
+    return;         // Error: Unexpected newline before return statement.
```

Comment:
We are no longer putting comments in showing the error message. Can you remove them?